src/components/TextInput/BaseTextInput.js

```diff
@@ -18,384 +18,369 @@ import getSecureEntryKeyboardType from '../../libs/getSecureEntryKeyboardType';
 import FormHelpMessage from '../FormHelpMessage';
 import isInputAutoFilled from '../../libs/isInputAutoFilled';
-import * as Pressables from '../Pressable';
-
-const PressableWithoutFeedback = Pressables.PressableWithoutFeedback;
-class BaseTextInput extends Component {
-    constructor(props) {
-        super(props);
-
-        const value = props.value || props.defaultValue || '';
-        const activeLabel = props.forceActiveLabel || value.length > 0 || Boolean(props.prefixCharacter);
-
-        this.state = {
-            isFocused: false,
-            labelTranslateY: new Animated.Value(activeLabel ? styleConst.ACTIVE_LABEL_TRANSLATE_Y : styleConst.INACTIVE_LABEL_TRANSLATE_Y),
-            labelScale: new Animated.Value(activeLabel ? styleConst.ACTIVE_LABEL_SCALE : styleConst.INACTIVE_LABEL_SCALE),
-            passwordHidden: props.secureTextEntry,
-            textInputWidth: 0,
-            prefixWidth: 0,
-            selection: props.selection,
-            height: variables.componentSizeLarge,
-
-            // Value should be kept in state for the autoGrow feature to work - https://github.com/Expensify/App/pull/8232#issuecomment-1077282006
-            value,
-        };
-
-        this.input = null;
-        this.isLabelActive = activeLabel;
-        this.onPress = this.onPress.bind(this);
-        this.onFocus = this.onFocus.bind(this);
-        this.onBlur = this.onBlur.bind(this);
-        this.setValue = this.setValue.bind(this);
-        this.togglePasswordVisibility = this.togglePasswordVisibility.bind(this);
-        this.dismissKeyboardWhenBackgrounded = this.dismissKeyboardWhenBackgrounded.bind(this);
-        this.storePrefixLayoutDimensions = this.storePrefixLayoutDimensions.bind(this);
-    }
-
-    componentDidMount() {
-        if (this.props.disableKeyboard) {
-            this.appStateSubscription = AppState.addEventListener('change', this.dismissKeyboardWhenBackgrounded);
+import PressableWithoutFeedback from '../Pressable/PressableWithoutFeedback';
+
+function BaseTextInput(props) {
+    const inputValue = props.value || props.defaultValue || '';
+    const initialActiveLabel = props.forceActiveLabel || inputValue.length > 0 || Boolean(props.prefixCharacter);
+
+    const [isFocused, setIsFocused] = useState(false);
+    const labelTranslateY = useRef(new Animated.Value(initialActiveLabel ? styleConst.ACTIVE_LABEL_TRANSLATE_Y : styleConst.INACTIVE_LABEL_TRANSLATE_Y)).current;
+    const labelScale = useRef(new Animated.Value(initialActiveLabel ? styleConst.ACTIVE_LABEL_SCALE : styleConst.INACTIVE_LABEL_SCALE)).current;
+    const [passwordHidden, setPasswordHidden] = useState(props.secureTextEntry);
+    const [textInputWidth, setTextInputWidth] = useState(0);
+    const [textInputHeight, setTextInputHeight] = useState();
+    const [prefixWidth, setPrefixWidth] = useState(0);
+    const [height, setHeight] = useState(variables.componentSizeLarge);
+    const [width, setWidth] = useState();
```

Comment:
can we clean up a little bit - move useState items under each other, and after place useRefs item? it will be easier to read

Reply:
here we go 😊